compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1552,6 +1564,81 @@ private void defineMembers(List<BLangTypeDefinition> typeDefNodes, SymbolEnv pkg
         }
     }
 
+    private void validateReadOnlyIntersectionTypeDefinitions(List<BLangTypeDefinition> typeDefNodes) {
+        List<BType> loggedTypes = new ArrayList<>();
+
+        for (BLangTypeDefinition typeDefNode : typeDefNodes) {
+            BLangType typeNode = typeDefNode.typeNode;
+            NodeKind kind = typeNode.getKind();
+            if (kind == NodeKind.INTERSECTION_TYPE_NODE) {
+                BType currentType = typeNode.type;
+
+                if (currentType.tag != TypeTags.INTERSECTION) {
```

Comment:
Isn't the type an intersection type if the type node is an intersection type node?

Reply:
This can be `symTable.noType` if a member type could not be resolved.

e.g., where `Foo` is not defined
```ballerina
type MyType Foo & readonly;
```